spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualAckSample.java

```diff
@@ -0,0 +1,71 @@
+package io.awspring.cloud.sqs.sample;
+
+import io.awspring.cloud.sqs.annotation.SqsListener;
+import io.awspring.cloud.sqs.config.SqsMessageListenerContainerFactory;
+import io.awspring.cloud.sqs.listener.acknowledgement.Acknowledgement;
+import io.awspring.cloud.sqs.listener.acknowledgement.AcknowledgementOrdering;
+import io.awspring.cloud.sqs.listener.acknowledgement.handler.AcknowledgementMode;
+import io.awspring.cloud.sqs.operations.SqsTemplate;
+import io.awspring.cloud.sqs.operations.TemplateAcknowledgementMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.Message;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
+
+import java.time.Duration;
+import java.util.UUID;
+
+@Configuration
+public class SqsManualAckSample {
+
+    public static void main(String[] args) {
+        SpringApplication.run(SqsManualAckSample.class);
+    }
+
+    public static final String NEW_USER_QUEUE = "new-user-queue";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SqsManualAckSample.class);
+
+    @Bean
+    public ApplicationRunner sendMessageToQueue(SqsTemplate sqsTemplate) {
+        LOGGER.info("Sending message");
+        return args -> sqsTemplate.send(to -> to.queue(NEW_USER_QUEUE)
+                .payload(new User(UUID.randomUUID(), "John"))
+        );
+    }
+
+    @Bean
+    public SqsTemplate sqsTemplate(SqsAsyncClient sqsAsyncClient) {
+        return SqsTemplate.builder()
+                .sqsAsyncClient(sqsAsyncClient)
+                .configure(options -> options.acknowledgementMode(TemplateAcknowledgementMode.MANUAL))
+                .build();
+    }
+
+    @SqsListener(NEW_USER_QUEUE)
+    public void listen(Message<User> message) {
+        LOGGER.info("Received message {}", message.getPayload());
+        Acknowledgement.acknowledge(message);
```

Comment:
Maybe add a "Message {} acknowledged" log here?

Reply:
Or, actually, if we're doing the AckResult thing, no need for this log I suggested, we can just keep the log in the result.

Reply:
I will add the log after Acknowledgment, and make it simple... I think that we can add in the new [documentation](https://awspring.io/learn/quick-start/quick-start/) some how-to guides for each real-world problem (based on [diatáxis framework](https://diataxis.fr/)). 

> How-to guides are directions that take the reader through the steps required to solve a real-world problem. How-to guides are goal-oriented.

Reply:
I went back and added the AcknowledgmentResultCallback :smile: